File: projects/spring-boot-restv2/demo/src/main/java/com/sourabh/demo/model/Transaction.java
package com.sourabh.demo.model;

import io.swagger.annotations.ApiModel;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.stereotype.Component;

@Document(collection = "transactions")
@Component
@ApiModel(description = "Details about the transactions.")
public class Transaction {
    @Id
    private int id;
    private Customer customer;
    private float amount;

    public Transaction(){}

    public Transaction(int id, Customer customer, float amount) {
        this.id = id;
        this.customer = customer;
        this.amount = amount;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Customer getCustomer() {
        return customer;
    }

    public void setCustomer(Customer customer) {
        this.customer = customer;
    }

    public float getAmount() {
        return amount;
    }

    public void setAmount(float amount) {
        this.amount = amount;
    }
}

File: projects/spring-boot-rest-kt/hotel-reservation/src/main/kotlin/com/sourabh/hotelreservation/model/Transaction.kt
package com.sourabh.hotelreservation.model

import io.swagger.annotations.ApiModel
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.stereotype.Component



@Document(collection = "transactionK")
@ApiModel(description = "Details about the Customer.")
data class Transaction(@Id var id:Int=0,
                  var customer: Customer,
                  var amount:Float ) {
}